hathor/manager.py

```diff
@@ -516,19 +517,33 @@ def generate_parent_txs(self, timestamp: Optional[float]) -> 'ParentTxs':
         This method tries to return a stable result, such that for a given timestamp and storage state it will always
         return the same.
         """
-        if timestamp is None:
-            timestamp = self.reactor.seconds()
-        can_include_intervals = sorted(self.tx_storage.get_tx_tips(timestamp - 1))
-        assert can_include_intervals, 'tips cannot be empty'
-        max_timestamp = max(int(i.begin) for i in can_include_intervals)
-        must_include: List[bytes] = []
-        assert len(can_include_intervals) > 0, f'invalid timestamp "{timestamp}", no tips found"'
-        if len(can_include_intervals) < 2:
-            # If there is only one tip, let's randomly choose one of its parents.
-            must_include_interval = can_include_intervals[0]
-            must_include = [must_include_interval.data]
-            can_include_intervals = sorted(self.tx_storage.get_tx_tips(must_include_interval.begin - 1))
-        can_include = [i.data for i in can_include_intervals]
+        # get all that are before "timestamp"
```

Comment:
This code seems good to me but it depends on the `iter_mempool_tips`. Maybe add as `_generate_parent_txs_new`, so we can run the tests and compare both of them. We can make it default later.

Reply:
Makes sense.

Reply:
Great! Thanks!